src/Block/AdminStatsBlockService.php

```diff
@@ -49,11 +49,15 @@ public function execute(BlockContextInterface $blockContext, Response $response
         $filters = $blockContext->getSetting('filters');
 
         if (!isset($filters['_per_page'])) {
-            $filters['_per_page'] = ['value' => $blockContext->getSetting('limit')];
+            $filters['_per_page'] = $blockContext->getSetting('limit');
         }
 
         foreach ($filters as $name => $data) {
-            $datagrid->setValue($name, $data['type'] ?? null, $data['value']);
+            if (\is_array($data)) {
```

Comment:
Sorry for not agreeing these changes, I think this way could be better:
```php
if (\is_scalar($data)) {
    $datagrid->setSimpleValue($name, $data);
} else {
    $datagrid->setValue($name, $data['type'] ?? null, $data['value']);
}
```
But I don't really feel comfortable adding a method just for a specific value type. IMO, it's still hiding the real issue here.

Reply:
In AdminStatsBlockService, the only way to push data to Datagrid is with `setValue`...

Reply:
Yes, that's why I think we should keep that responsibility narrowed to `setValue()`, triggering any required check or transformation there.